src/CodeGen_Internal.cpp

```diff
@@ -385,18 +385,20 @@ bool get_md_string(llvm::Metadata *value, std::string &result) {
     return false;
 }
 
-void get_target_options(const llvm::Module &module, llvm::TargetOptions &options, std::string &mcpu, std::string &mattrs) {
+void get_target_options(const llvm::Module &module, llvm::TargetOptions &options,
+                        std::string &mcpu, std::string &mattrs, const Halide::Target &target) {
     bool use_soft_float_abi = false;
     get_md_bool(module.getModuleFlag("halide_use_soft_float_abi"), use_soft_float_abi);
     get_md_string(module.getModuleFlag("halide_mcpu"), mcpu);
     get_md_string(module.getModuleFlag("halide_mattrs"), mattrs);
 
+    bool strict_fp = target.has_feature(Target::StrictFP);
```

Comment:
I think this new option should be also embedded in the module as metadata (halide_strict_fp or something) like the other options above (particularly halide_use_soft_float_abi which is also a bool). That way, you don't need to plumb around the target (which is most of the changes in this PR :), the llvm::Module is self contained.

Reply:
@abadams Any opinion on this?

Reply:
sgtm. It is nice to have a self-contained llvm::Module - i.e. one that tells us what TargetOptions to use for it. The consistency argument is also a strong one